iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -131,6 +129,7 @@ public void AddOrUpdateComponentProperties(string componentName, IDictionary<str
         /// </para>
         /// </remarks>
         /// <param name="properties">A collection of properties to add or update.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="properties"/> is <c>null</c>.</exception>
```

Comment:
need to inherit only 1 exception under `AddRootProperty`, so we'll need to specify it explicitly instead.